components/web_discovery/browser/request_queue.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_REQUEST_QUEUE_H_
+#define BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_REQUEST_QUEUE_H_
+
+#include "base/functional/callback.h"
+#include "base/memory/raw_ptr.h"
+#include "base/timer/timer.h"
+#include "base/values.h"
+#include "net/base/backoff_entry.h"
+
+class PrefService;
+
+namespace web_discovery {
+
+// Persists and schedules requests on randomized intervals within
+// an interval range. If request failures exceed the threshold defined in
+// `max_retries`, the request will be dropped from the list. If a persisted
+// request age exceeds `request_max_age`, the request will be dropped.
+class RequestQueue {
```

Comment:
We need tests for `RequestQueue`.

Reply:
`RequestQueue` is tested via both `WebDiscoveryDoubleFetcherTest` and `WebDiscoveryReporterTest`

Reply:
Yeah, but we still should have individual tests for this class in particular, decoupled from the other uses, even if only to demonstrate its invariants.